### PR TITLE
chore: fix deprecated IntegrationTest

### PR DESCRIPTION
**Pull request by Roiocam**
IntegrationTest currently not working on my IDEA, It keeps complaining about NoClassFound.

After following the changes in the sbt release notes guidance, it works fine on my IDEA: https://eed3si9n.com/sbt-1.9.0

### CODE REVIEW

**Comment by pjfanning** [comment]
Any idea why the Check Code Style CI build is failing when it is looking for pekko-persistence-jdbc-core? This jar seems never to have been published before.

File: build.sbt [hunk 1]
```diff
@@ -33,27 +33,37 @@ lazy val core = project
   .in(file("core"))
   .enablePlugins(MimaPlugin, ReproducibleBuildsPlugin)
   .disablePlugins(SitePlugin)
-  .configs(IntegrationTest.extend(Test))
-  .settings(Defaults.itSettings)
   .settings(
-    name := "pekko-persistence-jdbc",
+    name := "pekko-persistence-jdbc-core",
```
Comment (pjfanning) [inline build.sbt:37]: this is breaking our build

Comment (Roiocam) [reply build.sbt]: Yes, I just investigated here too.

**Review by pjfanning — APPROVED** [review]
lgtm

**Pull request merged**